### ISSUE TITLE
[Improvement] Pipeline Style

### ISSUE THREAD

**Issue by guidiego** [jusbrasil/kube-render#16]
**Problems:** 
- Today we look all template-dir or template-url and compile all files (includes macro's one)
- Today we didn't have a logical sequence between compilation and if one day we need to make some complex compile or have files that depend on another file we can't throw exceptions or make nothing related to it.
- Our templates need to have tons of logic to prevent errors about missing params to ignore a template that doesn't fulfill the requirements

**Solution:** 
 -  I want to add a `-p` / `--pipe` to `kube-render` this will allow user to send me what files they want to compile in a sequence
 - When I ask for a file and this one couldn't be rendered for any reason I want to throw it.
 - Improve kube-template adding `docs`

## Usage
`kube-render -t xxxx -c xxxx -p deployment service --verbose --apply`

**Comment by andrewsmedina** [comment]
In which cases the compile in a order is important?

**Comment by guidiego** [comment]
@andrewsmedina I don't think that we have a case right now, that's a hypothetical question for future features. But point 1 and 2 are what I and @andersonba discuss a lot of.
